virtualizarr/tests/test_manifests/test_store.py

```diff
@@ -33,19 +30,64 @@ def filepath(tmpdir):
         filepath,
         b"\x01\x02\x03\x04\x05\x06\x07\x08\x09\x10\x11\x12\x13\x14\x15\x16",
     )
-    return f"{tmpdir}/{filepath}"
+    chunk_dict = {
+        "0.0": {"path": f"file://{tmpdir}/{filepath}", "offset": 0, "length": 4},
+        "0.1": {"path": f"file://{tmpdir}/{filepath}", "offset": 4, "length": 4},
+        "1.0": {"path": f"file://{tmpdir}/{filepath}", "offset": 8, "length": 4},
+        "1.1": {"path": f"file://{tmpdir}/{filepath}", "offset": 12, "length": 4},
+    }
+    manifest = ChunkManifest(entries=chunk_dict)
+    chunks = (1, 4)
+    shape = (2, 8)
+    array_metadata = array_v3_metadata(shape=shape, chunks=chunks)
+    manifest_array = ManifestArray(metadata=array_metadata, chunkmanifest=manifest)
+    manifest_group = ManifestGroup(
+        {"foo": manifest_array, "bar": manifest_array}, attributes={"Zarr": "Hooray!"}
+    )
+    return ManifestStore(
+        stores={"file://": obs.store.LocalStore()}, manifest_group=manifest_group
+    )
 
 
-@requires_obstore
 @pytest.fixture()
-def manifest_store(filepath, array_v3_metadata):
+def s3_store(minio_bucket, array_v3_metadata):
     import obstore as obs
 
+    store = obs.store.S3Store(
+        minio_bucket["bucket"],
+        aws_endpoint=minio_bucket["endpoint"],
+        access_key_id=minio_bucket["username"],
+        secret_access_key=minio_bucket["password"],
+        virtual_hosted_style_request=False,
+        client_options={"allow_http": True},
+    )
+    filepath = "data.tmp"
+    obs.put(
+        store,
+        filepath,
+        b"\x01\x02\x03\x04\x05\x06\x07\x08\x09\x10\x11\x12\x13\x14\x15\x16",
+    )
     chunk_dict = {
-        "0.0": {"path": f"file://{filepath}", "offset": 0, "length": 4},
-        "0.1": {"path": f"file://{filepath}", "offset": 4, "length": 4},
-        "1.0": {"path": f"file://{filepath}", "offset": 8, "length": 4},
-        "1.1": {"path": f"file://{filepath}", "offset": 12, "length": 4},
+        "0.0": {
+            "path": f"s3://{minio_bucket['bucket']}/{filepath}",
+            "offset": 0,
+            "length": 4,
+        },
+        "0.1": {
+            "path": f"s3://{minio_bucket['bucket']}/{filepath}",
+            "offset": 4,
+            "length": 4,
+        },
+        "1.0": {
+            "path": f"s3://{minio_bucket['bucket']}/{filepath}",
+            "offset": 8,
+            "length": 4,
+        },
+        "1.1": {
+            "path": f"s3://{minio_bucket['bucket']}/{filepath}",
+            "offset": 12,
+            "length": 4,
+        },
     }
     manifest = ChunkManifest(entries=chunk_dict)
```

Comment:
Is there a way to split this fixture up a bit, to more clearly delineate the creation of data in a (minio) s3-like obstore from the creation of the `ManifestStore` object which contains the byte references pointing to it?

Reply:
good call, done in [`874c6df` (#507)](https://github.com/zarr-developers/VirtualiZarr/pull/507/commits/874c6df4fa26a1271720f6f6a1c04575429d13ab)